sklearn/metrics/tests/test_ranking.py

```diff
@@ -1100,6 +1109,24 @@ def test_precision_recall_curve_toydata():
             )
 
 
+def test_precision_recall_curve_drop_intermediate():
+    # Test that drop_intermediate drops the correct thresholds
+    y_true = [0, 0, 0, 0, 1, 1]
+    y_score = [0.0, 0.2, 0.5, 0.6, 0.7, 1.0]
+    precision, recall, thresholds = precision_recall_curve(
+        y_true, y_score, drop_intermediate=True
+    )
+    assert_array_almost_equal(thresholds, [0.0, 0.7, 1.0])
+
+    # Test dropping thresholds with repeating scores
+    y_true = [0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1]
+    y_score = [0.0, 0.1, 0.6, 0.6, 0.7, 0.8, 0.9, 0.6, 0.7, 0.8, 0.9, 0.9, 1.0]
+    precision, recall, thresholds = precision_recall_curve(
+        y_true, y_score, drop_intermediate=True
+    )
+    assert_array_almost_equal(thresholds, [0.0, 0.6, 0.7, 0.8, 0.9, 1.0])
+
```

Comment:
I think that we should add some corner cases on the limit, e.g. only a single class or all TP or all FP, etc. to check if we don't report something odd compared to the undropped version.